sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFileFormat.scala

```diff
@@ -718,7 +722,9 @@ object ParquetFileFormat extends Logging {
   }
 
   /**
-   * ParquetFileFormat.prepareWrite calls this function to initialize `redirectParquetLogsViaSLF4J`.
+   * The `ParquetFileFormat` constructor calls this method to ensure that Parquet library log
+   * output is redirected through the SLF4J JUL bridge handler. This method is a no-op because we
+   * only require that `redirectParquetLogsViaSLF4J` is initialized. This accomplishes that task.
    */
-  def redirectParquetLogs(): Unit = {}
+  private def ensureParquetLogRedirection(): Unit = {}
```

Comment:
I suppose it's no worse than the existing setup, though surely there's a better way to get once-per-class execution. Would you try one more thing: 

`private def ensureParquetLogRedirection(): Unit = redirectParquetLogsViaSLF4J`

_That_ would at least connect the things that are supposed to be connected though it's all a no-op and would look less wonky to me.

If that still generates a warning OK let's merge this.